### DISCUSSION TITLE
Question - Possibility to track the download time/speed

### DISCUSSION THREAD

**Discussion by minedun6** [spatie/ssh#44] [Ideas]
First and foremost, thanks to all the spatie team and contributors which make our work on products much easier.

What I would like to know is if it's possible to expose the donwload speed, ellapsed time, remaining time, size of download file of the download function ?

Maybe something like this to better illustrate my question.
![Capture d’écran 2021-03-19 à 11 24 25](https://user-images.githubusercontent.com/4510058/111766368-befa7b00-88a5-11eb-8664-3b85bfccc546.png)

**Comment by freekmurze** [comment]
Hi, I don't have a need for this, some I'm not going to invest time. 

If this can be added in a clean, testable way, feel free to send a PR.

**Comment by minedun6** [comment]
Thank you for moving the issue so that more people can help/point me in the correct direction.
Maybe I can re-phrase the question a bit better.
I'm looking for a way to do this kind of behaviour. I'm looking for pointers ^^".
